recipes/libsecp256k1-py-bindings/meta.yaml

```diff
@@ -0,0 +1,60 @@
+{% set name = "libsecp256k1-py-bindings" %}
+{% set version = "0.4.0" %}
+
+package:
+  name: {{ name }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/MementoRC/coincurve/archive/refs/tags/v{{ version }}.tar.gz
+  sha256: ef07d2e575cbb2df14a1005b4cb329722049319567d20ac3d9e834630941d9eb
+
+build:
+  number: 0
+  script:  {{ PYTHON }} -m pip install . --use-pep517 -vvv
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - pkg-config  # [not win]
+  host:
+    - cffi >=1.3.0
+    - libsecp256k1
+    - pip
+    - pkgconfig
+    - python
+  run:
+    - asn1crypto
+    - {{ pin_compatible('cffi') }}
+    - python
+
+test:
+  imports:
+    - libsecp256k1_py_bindings
+  source_files:
+    - tests/
+  requires:
+    - pip
+    - pytest
+    - pytest-benchmark
+  commands:
+    - pip check
+    - pytest -v tests
+
+about:
+  home: https://github.com/ofek/coincurve
+  summary: Cross-platform Python CFFI bindings for libsecp256k1
+  doc_url: https://ofek.dev/coincurve/
+  description: |
+    This library provides well-tested Python bindings for libsecp256k1,
+    the heavily optimized C library used by Bitcoin Core for operations
+    on the elliptic curve secp256k1.
+  license: MIT OR APACHE-2.0
+  license_file:
+    - LICENSE-MIT
+    - LICENSE-APACHE
```

Comment:
Wait does coincurve need a new release?

Reply:
Yes. I believe that the latest release of coincurve (v18) links to a version of libsekcp256k1 that is too old for @MementoRC 's use case. However, I'm not sure whether even the main branch of coincurve contains a new enough version since that version is v0.3 and this PR contains a fork of coincurve which links to v0.4. Please coordinates amongst yourselves.

Reply:
@ofek, A short answer is to provide a release with the recent merge so that the conda package could rely upon a release tag rather than a commit. But still, that would not resolve the key issue

Based on my understanding the main issue is that, because it repackages the `libsecp256k1` library from bitcoin-core, `coincurve` cannot be a clean conda package. The solution I propose is to provide the binding package that relies upon a separate conda package (C lib) that provides `libsecp256k1` - However, that package does not exist in Pypi and thus it would require you to create a backward incompatibility with how you deliver your package (binding+C lib)

The conundrum is that, a conda recipe cannot change the upstream resource, so all the patches that I made to provide a conda packaging of coincurve are not acceptable

Reply:
Oh, I see what you mean. That's a tough situation. Let me know if I can help.

Reply:
> But still, that would not resolve the key issue

It would resolve the issue because coincurve will [detect and link dynamically to existing system libraries of libsecp256k1](https://github.com/ofek/coincurve/blob/2dfea673835e524ffaeb11dc1c2aecaf23188195/setup_support.py#L84). So repackagaging is not the issue because coincurve can dynamically link to a system provided libsecp256k1 (the static linking / repackaging is already optional).

The issue is whether the latest stable release of coincurve can link to a stable release of libsecp256k1.

If @ofek makes a new release for coincurve v19 that is ABI compatible with libsecp256k1 v0.3.2 then both coincurve and libsecp256k1 will be build from stable releases. That should solve the problem as far as this channel is concerned.

Reply:
@MementoRC, please see comment above.

Reply:
@carterbox but how would libsecp256k1 be made available in a non-conda environment? Or are you saying that it is not necessary to remove the repackaging, but create a conda package that requires the `libsecp256k1` conda package?
As-is, the _has_system_lib will not work as the path is hard-coded
@ofek, I plan to look into providing a PR for version 0.4.0

Reply:
> how would libsecp256k1 be made available in a non-conda environment?

That's irrelevant? Conda packages are only for conda environments.

> Is not necessary to remove the repackaging, but create a conda package that requires the libsecp256k1 conda package?

Yes. I'm pretty sure that coincurve [uses externally provided libsecp256k1 by default](https://github.com/ofek/coincurve/blob/1dd0675add2da67ad9e6acfc47eed041066f2ea1/setup.py#L134) and only revends if it cannot find an externally provided libsecp256k1. You just need to add the libsec256k1 package to the host dependencies of the coincurve recipe and ensure that the coincurve setup script can find it correctly; this probably requires no/very few patches.

You can probably even use v0.4.0 of libsecp256k1 because it is ABI compatible with v0.3.x.